### DISCUSSION TITLE
Federating public registries

### DISCUSSION THREAD

**Discussion by selfagency** [verdaccio/verdaccio#2525] [Ideas]
Uplinks are a great feature. But I would love to see npm become wholly decentralized and replaced with a distributed package system over which open source contributors have more control. This could be done by federating public Verdaccio registries using a protocol like ActivityPub. Whenever a package is modified/added/etc. it can be pushed as an ActivityPub event that is distributed across the network, making other registries aware of the change. This could be augmented by an indexing service that maps relations between packages and their host registries, public instances of which could be voluntarily hosted.

**Comment by juanpicado** [comment]
Have you heard of 👇🏼 ? 
https://github.com/entropic-dev/entropic
https://www.youtube.com/watch?v=MO8hZlgK5zc

**Reply by selfagency** [reply]
I have but I thought It was kind of dead and AFAIK doesn't have the web frontend Verdaccio has
